maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/GoalRenderer.java

```diff
@@ -0,0 +1,532 @@
+        renderAttribute(descriptor.getExecuteLifecycle(), "executeLifecycle");
+        renderAttribute(descriptor.isOnlineRequired(), "onlineRequired");
+        renderAttribute(!descriptor.isInheritedByDefault(), "notInheritedByDefault");
+
+        sink.list_();
+    }
+
+    private void renderAttribute(boolean condition, String attributeKey) {
+        renderAttribute(condition, attributeKey, Optional.empty());
+    }
+
+    private void renderAttribute(String conditionAndCodeArgument, String attributeKey) {
+        renderAttribute(
+                StringUtils.isNotEmpty(conditionAndCodeArgument),
+                attributeKey,
+                Optional.ofNullable(conditionAndCodeArgument));
+    }
+
+    private void renderAttribute(boolean condition, String attributeKey, Optional<String> codeArgument) {
+        if (condition) {
+            sink.listItem();
+            linkPatternedText(getI18nString(attributeKey));
+            if (codeArgument.isPresent()) {
+                text(": ");
+                sink.inline(Semantics.CODE);
+                sink.text(codeArgument.get());
+                sink.inline_();
+            }
+            text(".");
+            sink.listItem_();
+        }
+    }
+
+    private void renderParameterOverviewTable(String title, Iterator<Parameter> parameters) {
+        // don't emit empty tables
+        if (!parameters.hasNext()) {
+            return;
+        }
+        startSection(title);
+        startTable();
+        tableHeader(new String[] {
+            getI18nString("parameter.name.header"),
+            getI18nString("parameter.type.header"),
+            getI18nString("parameter.since.header"),
+            getI18nString("parameter.description.header")
+        });
+        while (parameters.hasNext()) {
+            renderParameterOverviewTableRow(parameters.next());
+        }
+        endTable();
+        endSection();
+    }
+
+    private void renderTableCellWithCode(String text) {
+        renderTableCellWithCode(text, Optional.empty());
+    }
+
+    private void renderTableCellWithCode(String text, URI link) {
+        renderTableCellWithCode(text, Optional.of(link));
+    }
+
+    private void renderTableCellWithCode(String text, Optional<URI> link) {
+        sink.tableCell();
+        if (link.isPresent()) {
+            sink.link(link.get().toString(), null);
+        }
+        sink.inline(Semantics.CODE);
+        sink.text(text);
+        sink.inline_();
+        if (link.isPresent()) {
+            sink.link_();
+        }
+        sink.tableCell_();
+    }
+
+    private void renderParameterOverviewTableRow(Parameter parameter) {
+        sink.tableRow();
+        // name
+        try {
+            // link to appropriate section
+            renderTableCellWithCode(
+                    format("parameter.name", parameter.getName()),
+                    new URI(null, null, HtmlTools.encodeId(parameter.getName())));
```

Comment:
Why encode when you use the URI class, that should abstract it away no? I have the feeling that the usage of `URI` is absolutely not necessary anymore here. It maybe sufficient to pass around opaque valid strings.

Reply:
Yes, the Sink impl is responsible for encoding the URL (encoding the id is necessary so that the attribute of the link and the one emitted for the anchor match). Should be fixed in https://github.com/apache/maven-plugin-tools/pull/225/commits/f6dc03408e8346922c4f350b19aef18ee4653889.